<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>531</width>
    <height>581</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Colour space viewer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="colour_space_viewer.qrc">
    <normaloff>:/main/icon.png</normaloff>:/main/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Colour space</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QRadioButton" name="radioButton_rgb">
             <property name="text">
              <string>RGB</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_hsv">
             <property name="text">
              <string>HSV</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_hsl">
             <property name="text">
              <string>HSL</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_ycbcr">
             <property name="text">
              <string>YCbCr</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="radioButton_hcy">
             <property name="text">
              <string>HCY</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Selected colour:</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QLabel" name="label_title_b">
        <property name="text">
         <string>G</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="horizontalSlider_b">
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="pageStep">
         <number>16</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_value_b">
        <property name="minimumSize">
         <size>
          <width>40</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="2">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_title_a">
        <property name="text">
         <string>R</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="horizontalSlider_a">
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="pageStep">
         <number>16</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_value_a">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2">
     <widget class="ColourPlaneView" name="view_c">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>256</width>
        <height>256</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
      <property name="sceneRect">
       <rectf>
        <x>0.000000000000000</x>
        <y>0.000000000000000</y>
        <width>256.000000000000000</width>
        <height>256.000000000000000</height>
       </rectf>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="ColourPlaneView" name="view_b">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>256</width>
        <height>256</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
      <property name="sceneRect">
       <rectf>
        <x>0.000000000000000</x>
        <y>0.000000000000000</y>
        <width>256.000000000000000</width>
        <height>256.000000000000000</height>
       </rectf>
      </property>
     </widget>
    </item>
    <item row="3" column="2">
     <widget class="ColourPlaneView" name="view_a">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>256</width>
        <height>256</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
      <property name="sceneRect">
       <rectf>
        <x>0.000000000000000</x>
        <y>0.000000000000000</y>
        <width>256.000000000000000</width>
        <height>256.000000000000000</height>
       </rectf>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGraphicsView" name="colour_view">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Ignored">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOff</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="2">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_title_c">
        <property name="text">
         <string>B</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="horizontalSlider_c">
        <property name="maximum">
         <number>255</number>
        </property>
        <property name="pageStep">
         <number>16</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_value_c">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ColourPlaneView</class>
   <extends>QGraphicsView</extends>
   <header location="global">colourplaneview.hpp</header>
   <slots>
    <signal>mouseDragged(QPoint)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="colour_space_viewer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>horizontalSlider_c</sender>
   <signal>valueChanged(int)</signal>
   <receiver>label_value_c</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>509</x>
     <y>280</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_c</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_c(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_rgb</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>set_colourspace_rgb()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>54</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>36</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_hsv</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>set_colourspace_hsv()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>21</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_hsl</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>set_colourspace_hsl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>42</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>92</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_ycbcr</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>set_colourspace_ycbcr()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>115</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_a</sender>
   <signal>valueChanged(int)</signal>
   <receiver>label_value_a</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>572</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_b</sender>
   <signal>valueChanged(int)</signal>
   <receiver>label_value_b</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>572</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_b</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_b(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>161</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>564</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_a</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_a(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>485</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>565</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>radioButton_hcy</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>set_colourspace_hcy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view_a</sender>
   <signal>mouseDragged(QPoint)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_bc(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>515</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>528</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view_b</sender>
   <signal>mouseDragged(QPoint)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_ac(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>381</y>
    </hint>
    <hint type="destinationlabel">
     <x>264</x>
     <y>382</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>view_c</sender>
   <signal>mouseDragged(QPoint)</signal>
   <receiver>MainWindow</receiver>
   <slot>update_ab(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>164</y>
    </hint>
    <hint type="destinationlabel">
     <x>528</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>update_a(int)</slot>
  <slot>set_colourspace_rgb()</slot>
  <slot>set_colourspace_hsv()</slot>
  <slot>set_colourspace_hsl()</slot>
  <slot>set_colourspace_ycbcr()</slot>
  <slot>update_b(int)</slot>
  <slot>update_c(int)</slot>
  <slot>set_colourspace_hcy()</slot>
  <slot>update_ab(QPoint)</slot>
  <slot>update_ac(QPoint)</slot>
  <slot>update_bc(QPoint)</slot>
 </slots>
</ui>
